cscs-checks/microbenchmarks/likwid/likwid.py

```diff
@@ -0,0 +1,176 @@
+import reframe as rfm
+import reframe.utility.sanity as sn
+
+
+class BandwidthBase(rfm.RegressionTest):
+    def __init__(self):
+        super().__init__()
+
+        self.valid_prog_environs = ['PrgEnv-gnu']
+        self.build_system = 'Make'
+        self.build_system.flags_from_environ = False
+        self.sourcesdir = 'https://github.com/RRZE-HPC/likwid.git'
+        self.variables = {
+            'LD_LIBRARY_PATH': './lib:$LD_LIBRARY_PATH',
+            'PATH': './bin:./sbin:$PATH'
+        }
+
+        self.executable = 'bin/likwid-bench'
+
+        self.num_tasks = 1
+        self.num_tasks_per_core = 2
+        self.system_num_cpus = {
+            'daint:mc':  72,
+            'daint:gpu': 24,
+            'dom:mc':  72,
+            'dom:gpu': 24,
+        }
+        self.system_numa_domains = {
+            'daint:mc':  ['S0', 'S1'],
+            'daint:gpu': ['S0'],
+            'dom:mc':  ['S0', 'S1'],
+            'dom:gpu': ['S0'],
+        }
+
+        # Test each level at half capacity times nthreads per domain
+        self.system_cache_sizes = {
+            'daint:mc':  {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
+                          'memory': '1800MB'},
+            'daint:gpu': {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
+                          'memory': '1200MB'},
+            'dom:mc':    {'L1': '288kB', 'L2': '2304kB', 'L3': '23MB',
+                          'memory': '1800MB'},
+            'dom:gpu':   {'L1': '192kB', 'L2': '1536kB', 'L3': '15MB',
+                          'memory': '1200MB'},
+        }
+
+        self.maintainers = ['SK']
+        self.tags = {'diagnostic'}
+
+        bw_pattern = sn.extractsingle(r'MByte/s:\s*(?P<bw>\S+)',
+                                      self.stdout, 'bw',  float)
+
+        self.sanity_patterns = sn.assert_ge(bw_pattern, 0.0)
+        self.perf_patterns = {
+            'bandwidth': bw_pattern
+        }
+
+    def setup(self, partition, environ, **job_opts):
+        super().setup(partition, environ, **job_opts)
+
+        self.postbuild_cmd = ['make install PREFIX=%s INSTALL_CHOWN='
+                              '\'-g csstaff -o sebkelle\'' % self.stagedir]
+
+
+@rfm.required_version('>=2.16-dev0')
+@rfm.parameterized_test(*[[l,k] for l in ['L1', 'L2', 'L3']
+                        for k in ['load_avx', 'store_avx']],
+                        ['memory', 'load_avx'],
+                        ['memory', 'store_mem_avx'])
+class CPUBandwidth(BandwidthBase):
+    def __init__(self, mem_level, kernel_name):
+        super().__init__()
+
+        self.descr = 'CPU <- %s %s benchmark' % (mem_level, kernel_name)
+
+        self.valid_systems = ['daint:mc', 'daint:gpu', 'dom:gpu', 'dom:mc']
+
+        # the kernel to run in likwid
+        self.kernel_name = kernel_name
+        self.ml = mem_level
```

Comment:
I'd rather call `mem_level` the attribute than the function parameter. I had to get back to see what `ml` is. Better even, if you call both `mem_level`.